src/CoreWlan/CWKeychain.cs

```diff
@@ -0,0 +1,318 @@
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryFindWiFiPassword (CWKeychainDomain domain, NSData ssid, out NSString? password, out OSStatus status)
+		{
+			password = null;
+			NSStringRef passwordPtr = IntPtr.Zero;
+			status = CWKeychainFindWiFiPassword (domain, ssid.GetHandle (), out passwordPtr);
+			if (passwordPtr != IntPtr.Zero) {
+				password = Runtime.GetNSObject<NSString> (passwordPtr, false);
+			}
+			return status == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryFindWiFiPassword (CWKeychainDomain domain, NSData ssid, out NSString? password)
+			=> TryFindWiFiPassword (domain, ssid, out password, out var _);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainSetWiFiEAPIdentity (CWKeychainDomain domain, NSDataRef ssid, SecIdentityRef identity);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TrySetWiFiEAPIdentity (CWKeychainDomain domain, NSData ssid, SecIdentity? identity, out OSStatus status)
+		{
+			status = CWKeychainSetWiFiEAPIdentity (domain, ssid.GetHandle (), identity.GetHandle ()!);
+			return status == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TrySetWiFiEAPIdentity (CWKeychainDomain domain, NSData ssid, SecIdentity? identity)
+			=> TrySetWiFiEAPIdentity (domain, ssid, identity, out var _);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainSetWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSDataRef ssid, NSStringRef username, NSStringRef password);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TrySetWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, NSString? username, NSString? password, out OSStatus status)
+		{
+			status = CWKeychainSetWiFiEAPUsernameAndPassword (domain, ssid.GetHandle (), username.GetHandle ()!, password.GetHandle ()!);
+			return status == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TrySetWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, NSString? username, NSString? password)
+			=>  TrySetWiFiEAPUsernameAndPassword (domain, ssid, username, password, out var _);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10, 9)]
+#endif
+		public static bool TrySetWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, string? username, string? password, out OSStatus status)
+		{
+			status = CWKeychainSetWiFiEAPUsernameAndPassword (domain, ssid.GetHandle (), CFString.CreateNative (username), CFString.CreateNative (password));
```

Comment:
You must release the result from `CFString.CreateNative` after you're done with it.